### DISCUSSION TITLE
Ability to use session for identifying current tenant instead of route or domain

### DISCUSSION THREAD

**Discussion by andrewbroberg** [filamentphp/filament#13313] [Ideas]
Has this been a requested feature before?

Essentially would like the ability to be able to use the session instead of a route parameter or domain to identify the current tenant.

Happy to take a look at PR ing this myself if it's something that Filament could benefit from.

**Comment by andrewbroberg** [comment]
@danharrin Is this a feature you guys would like in filament? Happy to PR this myself if so.

**Reply by danharrin** [reply]
I'm unsure, it also affects routing and specifically tenant switching... what is the advantage for users? if they have multiple browser tabs open it can cause permission issues after switching

**Reply by andrewbroberg** [reply]
It's mostly preference around not having the tenant prefix/id in the URL.

**Reply by danharrin** [reply]
I feel like most of the "preference" options we give users in Filament don't degrade the user experience... what's the solution for ensuring users don't hit permission issues in other tabs when switching tenants?

**Comment by eduilson** [comment]
I needed this feature and ended up implementing. I'm sharing the link to the code I wrote. I do believe it's a valid functionality, especially if you want to keep the URL clean. However, it would indeed be necessary to consider cases where the user has other tabs open in the browser (maybe a reload to the home page? A JavaScript event to apply the changes, ...).

Anyway, for my use case, this piece of code worked well. And it might serve as a starting point for anyone who needs to implement this feature in the future.

[Example](https://gist.github.com/eduilson/c0ad57e5c96b6505d643c665e1103d92)
